### ISSUE TITLE
[Patch] sync_first, link_dest and multiple backup points [sf#4]

### ISSUE THREAD

**Issue by bebehei** [rsnapshot/sourceforge-issues#4]
**Reported by noredshadow on 2007-11-27 03:52 UTC**
This is a patch that I use to fix this bug: https://sourceforge.net/tracker/index.php?func=detail&aid=1643701&group\_id=88546&atid=587076

A quick recap:
If you set both sync_first and link_dest and then you try to backup multiple directories, then --link-dest will only be added to the rsync call for the first directory.  This patch fixes the issue.

I tried to attach it to the bug directly, but I either couldn't or I was stupid and just copied and pasted it.

**Comment by bebehei** [comment]
**Commented by noredshadow on 2007-11-27 03:52 UTC**
Patch that fixes bug 1643701
Attached filersnapshot_link_dest.patch:

```
--- rsnapshot.orig  2007-05-25 03:40:45.000000000 -0400
+++ rsnapshot   2007-05-26 14:26:20.000000000 -0400
@@ -122,6 +122,7 @@
 my $do_configtest  = 0; # parse config file and exit
 my $one_fs         = 0; # one file system (don't cross partitions within a backup point)
 my $link_dest      = 0; # use the --link-dest option to rsync
+my $created_sync       = 0; # jgibson - Fix the sync and --link-dest bug

 # how much noise should we make? the default is 2
 #
@@ -3186,7 +3187,14 @@
    # (this may create the .sync dir, which is why we had to check for it above)
    #
    create_backup_point_dir($interval, $bp_ref);
-   
+   # jgibson - Did we create the sync dir ourself?
+   if ($sync_dir_was_present eq 0) {
+       if($interval eq 'sync') {
+           if ( -d "$config_vars{'snapshot_root'}/.sync" ) {
+               $created_sync = 1
+           }
+       }
+   }
    # check opts, first unique to this backup point, and then global
    #
    # with all these checks, we try the local option first, and if
@@ -3270,8 +3278,8 @@
            if ( -d "$config_vars{'snapshot_root'}/$interval_link_dest.$interval_num_link_dest/$$bp_ref{'dest'}" ) {

                # we don't use link_dest if we already synced once to this directory
-               if ($sync_dir_was_present) {
-                   
+               if ($sync_dir_was_present && ($created_sync eq 0)) {
+                   # jgibson: Only skip --link-dest if we didn't create the sync directory ourself.
                    # skip --link-dest, this is the second time the sync has been run, because the .sync directory already exists

                # default: push link_dest arguments onto cmd stack
```

**Comment by bebehei** [comment]
**Commented by djk20 on 2009-04-01 03:22 UTC**
There was a (slightly different) fix for this bug in rsnapshot 1.3.1.

**Comment by bebehei** [comment]
**Updated by djk20 on 2009-04-01 03:22 UTC**
- **status**: open --> closed-fixed
